stubs/docker/docker/types/daemon.pyi

```diff
@@ -1,6 +1,11 @@
-class CancellableStream:
+from typing import Generic, TypeVar
+from typing_extensions import Self
+
+_T_co = TypeVar("_T_co", covariant=True)
+
+class CancellableStream(Generic[_T_co]):
     def __init__(self, stream, response) -> None: ...
```

Comment:
Let's also annotate the `stream` parameter while we're here:

```suggestion
from collections.abc import Iterator
from typing import Generic, TypeVar
from typing_extensions import Self

_T_co = TypeVar("_T_co", covariant=True)

class CancellableStream(Generic[_T_co]):
    def __init__(self, stream: Iterator[_T_co], response) -> None: ...
```

I think `Iterator` is the correct type hint, as it looks like the implementation only ever calls `next()` on `self._stream`, which is what the `stream` parameter is stored as internally.

The reason why it's important to annotate this parameter at the same time as making the class generic, is that it means that type checkers will be able to infer that `foo` in the following snippet is of type `CancellableStream[bytes]`; otherwise, it would infer `CancellableStream[Any]` unless you gave it an explicit annotation:

```py
foo = CancellableStream(iter(b'foo'), some_response_object)
```

Reply:
I'm happy to do this. I didn't originally because my expectation is that end users of this library are unlikely to create `CancellableStream`s.

Reply:
Yeah, I see that; I debated whether it was worth being nitpicky about this here 😄 I still think it's probably good to do this anyway out of principle.

Reply:
I'm always happy to learn! And I find the reviews on this project amazing as you have to look into all the third-party projects.